lib/stripe/stripe_object.rb

```diff
@@ -194,19 +194,24 @@ def serialize_params(obj, options = {})
 
     protected
 
-    def metaclass
-      class << self; self; end
+    # A protected field is one that doesn't get an accessor assigned to it
+    # (i.e. `obj.public = ...`) and one which is not allowed to be updated via
+    # the class level `Model.update(id, { ... })`.
+    def self.protected_fields
+      []
     end
 
-    def protected_fields
-      []
+    def metaclass
+      class << self; self; end
     end
 
     def remove_accessors(keys)
-      f = protected_fields
+      # not available in the #instance_eval below
+      protected_fields = self.class.protected_fields
+
       metaclass.instance_eval do
         keys.each do |k|
-          next if f.include?(k)
+          next if protected_fields.include?(k)
```

Comment:
Sorry for the diff churn, but I renamed this because the old naming as absolutely atrocious.